### PR TITLE
Fix heightByRows resizing

### LINKED ISSUE #1510: Height by rows doesn't work in context menu demos
### Description
Setting `grid.heightByRows = true` for the grid in https://cdn.vaadin.com/vaadin-context-menu/4.2.0/demo/#context-menu-grid-demos expands the grid height but not all the rows get rendered. Could be an issue in the demo.

### Expected outcome
All the 100 items get rendered

### Actual outcome
Only the first ~25 items are rendered while the grid height is expanded beyond this

### Live Demo
https://cdn.vaadin.com/vaadin-context-menu/4.2.0/demo/#context-menu-grid-demos

### Steps to reproduce
1. Open https://cdn.vaadin.com/vaadin-context-menu/4.2.0/demo/#context-menu-grid-demos
2. Set `grid.heightByRows = true` with the help of developer tools

### Browsers Affected
Chrome at least

### PR DESCRIPTION
**Pull request by samiheikki**
Fixes #1510

---
This change is [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-grid/1515)

### CODE REVIEW

File: src/vaadin-grid.html [hunk 1]
```diff
@@ -714,6 +715,10 @@
       notifyResize() {
         super.notifyResize();
       }
+
+      _heightByRowsChanged() {
+        this.notifyResize();
```
Comment (web-padawan) [inline src/vaadin-grid.html]: Should we guard against switching from `undefined` to `false` (default value)?

File: test/resizing.html [hunk 2]
```diff
@@ -58,6 +58,16 @@
     </template>
   </test-fixture>
 
+  <test-fixture id="height-by-rows">
+      <template>
```
Comment (web-padawan) [inline test/resizing.html]: nit: incorrect indentation here and below

File: src/vaadin-grid.html [hunk 3]
```diff
@@ -389,7 +389,8 @@
           heightByRows: {
             type: Boolean,
             value: false,
-            reflectToAttribute: true
+            reflectToAttribute: true,
```
Comment (web-padawan) [inline src/vaadin-grid.html]: is the reflecting to attribute related to the fix? I'm wondering whether we need it as an attribute.

Comment (samiheikki) [reply src/vaadin-grid.html]: I assume it is needed as the attribute is used for styling: https://github.com/vaadin/vaadin-grid/blob/1d8c80a93e5da57f12df09a61f259c085df909d4/src/vaadin-grid-styles.html#L41

File: src/vaadin-grid.html (hunk 1 above)
Comment (samiheikki) [reply src/vaadin-grid.html]: Good point, we can avoide one unnecessary notifyResize. Fixed.

File: test/resizing.html (hunk 2 above)
Comment (samiheikki) [reply test/resizing.html]: Fixed.

**Pull request merged**